backend/routes/data.py

```diff
@@ -22,6 +23,59 @@ def send_audio_file(file_name):
     return send_from_directory(app.config["UPLOAD_FOLDER"], file_name)
 
 
+def validate_segmentation(segment):
+    """Validate the segmentation before accepting the annotation's upload from users
+    """
+    required_key = {"annotations", "end_time", "transcription"}
+
+    if set(required_key).issubset(segment.keys()):
+        return True
+    else:
+        return False
+
+
+def generate_segmentation(
+    annotations, transcription,
+    start_time, end_time, data_id=None, segmentation_id=None
+):
+    """Generate a Segmentation from the required segment information
+    """
+    if segmentation_id is None:
+        if data_id is None:
+            segmentation = Segmentation(
+                start_time=start_time, end_time=end_time
+            )
+        else:
+            segmentation = Segmentation(
+                data_id=data_id,
+                start_time=start_time, end_time=end_time
+            )
+    else:
+        segmentation = Segmentation.query.filter_by(
```

Comment:
There's actually a small bug here. If `start_time` and `end_time` are updated, we don't set that in this segment. Can you please add two setter methods to `Segmentation` and fix this?

Reply:
Will do.

Reply:
Please check the updated.